### ISSUE TITLE
Streaming example doesn't work with slower operations

### ISSUE THREAD

**Issue by josephbisaillon** [sbruno81/rets-client#45]
I noticed that if you use the streaming example and put a setTimeout(callback, 500) the application will eventually use up too much memory and give an exception. 

_FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory_

Is this normal behavior, are the transformations I am applying during the streaming process too much? I am trying to cast the strings ss different types before pasting it to my mongo. But, it seems that the transformations can't keep up with the stream, which eventually gains too much memory and crashes.

**Comment by zacronos** [comment]
Hmm.  I don't have that problem while using the streaming API, but I'm not trying to coerce any data in the process.  It is entirely possible part of the streaming code in rets-client needs to be changed to properly create backpressure.  I'll try to get to this at some point, but honestly my time right now is very limited.  If you are at all familiar with node streams and want to take a look, be my guest!

As a short-term workaround, you might consider saving the values as-is for now, and then looping over your data later to normalize the data (setting a flag at that time to mark it as ready).

**Comment by josephbisaillon** [comment]
Yup as a work around saving the values as it does work. I'm currently using that strategy, I'll dive into the code base a little to see if I can contribute to a fix.

**Comment by nmccready** [comment]
@josephbisaillon can you post an example of your work around? I currently just started having streaming issues that are causing network failures. 

In a nutshell my entire pipeline is working but it is only failing while it is in progress piping through express to the browser.

**Comment by nmccready** [comment]
@josephbisaillon nvm, please disregard. I talked to zacranos and he thinks this is unrelated.

**Comment by zacronos** [comment]
FYI, I've been giving the code some close scrutiny lately, and I've determined that indeed, the way I wrote the streaming code bypasses the normal backpressure mechanism.  A proper fix would require not just some refactoring of the existing code, but likely also a rewrite of the MultipartParser dependency.

There's at least one other reason to consider doing such a rewrite, so I'll probably do it eventually; I can't promise how soon it will be.  If someone else wants to consider taking this on, however, I'd be happy to help.
